src/assets/scripts/client/game/ScoreLogic.js

```diff
@@ -0,0 +1,274 @@
+     * @param aircraftModel {AircraftModel}
+     * @private
+     */
+    _onAirspaceExitWithoutClearance(aircraftModel) {
+        this._radioCall(aircraftModel, 'leaving radar coverage without proper clearance', AIRPORT_CONTROL_POSITION_NAME.DEPARTURE, true);
+        GameController.events_recordNew(GAME_EVENTS.NOT_CLEARED_ON_ROUTE);
+    }
+
+    /**
+     * @for ScoreLogic
+     * @method _scoreWind
+     * @param aircraftModel {AircraftModel}
+     * @param action {string}
+     */
+    _scoreWind(aircraftModel, action) {
+        const isWarning = true;
+        const wind = aircraftModel.getWind();
+
+        // TODO: these two if blocks could be done in a single switch statement
+        if (wind.cross >= 20) {
+            GameController.events_recordNew(GAME_EVENTS.EXTREME_CROSSWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with major crosswind`, isWarning);
+        } else if (wind.cross >= 10) {
+            GameController.events_recordNew(GAME_EVENTS.HIGH_CROSSWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with crosswind`, isWarning);
+        }
+
+        if (wind.head <= -10) {
+            GameController.events_recordNew(GAME_EVENTS.EXTREME_TAILWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with major tailwind`, isWarning);
+        } else if (wind.head <= -5) {
+            GameController.events_recordNew(GAME_EVENTS.HIGH_TAILWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with tailwind`, isWarning);
+        }
+    }
+
+    /**
+     * @for ScoreLogic
+     * @method _scoreRunwaySeparation
+     * @param aircraftModel {AircraftModel}
+     * @param runwayModel {RunwayModel}
+     * @param action {string}
+     */
+    _scoreRunwaySeparation(aircraftModel, runwayModel, action) {
+        const previousAircraft = this._aircraftController.findAircraftByCallsign(runwayModel.lastDepartedAircraftId);
+
+        if (previousAircraft) {
+            const actualDistance = nm_ft(aircraftModel.distanceToAircraft(previousAircraft));
+            const requiredDistance = aircraftModel.model.calculateRunwaySeparationDistanceInFeet(previousAircraft.model);
+
+            if (actualDistance < requiredDistance || previousAircraft.isOnGround()) {
+                const isWarning = true;
+
+                GameController.events_recordNew(GAME_EVENTS.NO_TAKEOFF_SEPARATION);
+                UiController.ui_log(`${aircraftModel.callsign} ${action} while another aircraft was using the same runway`, isWarning);
+            }
+        }
+    }
+
+    /**
+     * Display a waring and record an illegal glideslope intercept event
+     *
+     * @for ScoreLogic
+     * @method _penalizeLocalizerInterceptAltitude
+     * @param aircraftModel {AircraftModel}
+     */
+    _penalizeLocalizerInterceptAltitude(aircraftModel) {
+        if (aircraftModel.isAboveGlidepath()) {
```

Comment:
same here, reverse logic and return early

Reply:
done